File: exercises/exercise-2-outbox-java/src/main/java/io/miragon/example/adapter/in/zeebe/SendConfirmationMailWorker.java
package io.miragon.example.adapter.in.zeebe;

import io.miragon.example.adapter.process.generated.NewsletterSubscriptionProcessApi.TaskTypes;
import io.miragon.example.application.port.in.SendConfirmationMailUseCase;
import io.miragon.example.domain.OperationId;
import io.miragon.example.domain.SubscriptionId;
import io.camunda.client.annotation.JobWorker;
import io.camunda.client.annotation.Variable;
import io.camunda.client.api.response.ActivatedJob;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class SendConfirmationMailWorker {

    private static final Logger log = LoggerFactory.getLogger(SendConfirmationMailWorker.class);

    private final SendConfirmationMailUseCase useCase;

    public SendConfirmationMailWorker(SendConfirmationMailUseCase useCase) {
        this.useCase = useCase;
    }

    @JobWorker(type = TaskTypes.NEWSLETTER_SEND_CONFIRMATION_MAIL)
    public void sendConfirmationMail(
            ActivatedJob job,
            @Variable("subscriptionId") String subscriptionId
    ) {
        log.debug("Received Zeebe job to send confirmation mail: {}", subscriptionId);
        useCase.sendConfirmationMail(
                new SubscriptionId(UUID.fromString(subscriptionId)),
                new OperationId(subscriptionId + "-" + job.getElementId())
        );
    }
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/adapter/inbound/zeebe/newsletter/SendConfirmationMailWorker.kt
package io.miragon.example.adapter.inbound.zeebe.newsletter

import io.camunda.client.annotation.JobWorker
import io.camunda.client.annotation.Variable
import io.camunda.client.api.response.ActivatedJob
import io.miragon.example.adapter.process.generated.NewsletterSubscriptionProcessApi.TaskTypes
import io.miragon.example.application.port.inbound.newsletter.SendConfirmationMailUseCase
import io.miragon.example.domain.SubscriptionId
import mu.KotlinLogging
import org.springframework.stereotype.Component
import java.util.*

@Component
class SendConfirmationMailWorker(
    private val useCase: SendConfirmationMailUseCase
) {

    private val log = KotlinLogging.logger {}

    @JobWorker(type = TaskTypes.ACTIVITY_SEND_CONFIRMATION_MAIL)
    fun handle(job: ActivatedJob, @Variable subscriptionId: String) {
        log.debug { "Received job to send confirmation mail for subscriptionId: $subscriptionId" }
        useCase.sendConfirmationMail(SubscriptionId(UUID.fromString(subscriptionId)))
    }
}
